### PR TITLE
task #66 (sort of) 

### PR DESCRIPTION
**Pull request by 0xbahaa**

### CODE REVIEW

File: selecttool.cpp [hunk 1]
```diff
@@ -93,3 +93,18 @@ void SelectTool::mouseDoubleClickEvent(QGraphicsSceneMouseEvent *event)
...
+void SelectTool::keyPressEvent(QKeyEvent *event)
+{
+    qInfo("Detected a key press...");
+
+    //0x01000007 is the int for "Del"
+    if (event->key() == 0x01000007)
+    {
+        qInfo("Pressed Del");
+        if (selectedItem){
+            selectedItem->hide();
```
Comment (lordadamson) [inline selecttool.cpp]: I don't think hiding the item is a good idea. We need to actually delete it.

File: selecttool.cpp [hunk 2]
```diff
@@ -9,7 +9,7 @@ bool SelectTool::isAStroke(QGraphicsItem *item)
 SelectTool::SelectTool(Paper* paper)
 {
 	m_paper = paper;
-	selectedItem = nullptr;
+    //selectedItem = nullptr;
```
Comment (lordadamson) [inline selecttool.cpp]: why did you comment this out? it's important to initialize pointers to null.

File: README.md [hunk 3]
```diff
@@ -11,7 +11,10 @@ This project is based on the wonderful Qt framework.
 If you're on Windows download and install it here: https://www.qt.io/download-open-source/
 If you're on Linux use your package manager.
 ### Ubuntu
-`sudo apt install sudo apt-get install qt5-default qttools5-dev-tools` (I don't have ubuntu, let me know if there are more packages the needs to be installed please).
+`sudo apt install sudo apt-get install qt5-default qttools5-dev-tools` 
+You'll also need the OpenGL include files, so use these as well:
+`sudo apt-get install mesa-common-dev`
+`sudo apt-get install libgl1-mesa-dev libglu1-mesa-dev`
```
Comment (lordadamson) [inline README.md]: I'd rather have all the dependencies in one line. So maybe you could go like this: `sudo apt install qt5-default qttools5-dev-tools mesa-common-dev libgl1-mesa-dev libglu1-mesa-dev`

File: paper.h [hunk 4]
```diff
@@ -50,9 +50,11 @@ public slots:
 	void mouseMoveEvent(QGraphicsSceneMouseEvent *event) Q_DECL_OVERRIDE;
 	void mouseReleaseEvent(QGraphicsSceneMouseEvent *event) Q_DECL_OVERRIDE;
 	void mouseDoubleClickEvent(QGraphicsSceneMouseEvent *event) Q_DECL_OVERRIDE;
+    //void keyPressEvent(QKeyEvent *event) Q_DECL_OVERRIDE;
```
Comment (lordadamson) [inline paper.h]: remove this comment please

File: selecttool.cpp [hunk 5]
```diff
@@ -93,3 +93,18 @@ void SelectTool::mouseDoubleClickEvent(QGraphicsSceneMouseEvent *event)
 
 	//m_paper->graphicsSceneDoubleClickEvent(event);
 }
+
+void SelectTool::keyPressEvent(QKeyEvent *event)
+{
+    qInfo("Detected a key press...");
+
+    //0x01000007 is the int for "Del"
+    if (event->key() == 0x01000007)
```
Comment (lordadamson) [inline selecttool.cpp]: replace that with Qt::Key_Delete. Would be much more readable this way.

File: selecttool.cpp [hunk 6]
```diff
@@ -93,3 +93,18 @@ void SelectTool::mouseDoubleClickEvent(QGraphicsSceneMouseEvent *event)
...
+    {
+        qInfo("Pressed Del");
```
Comment (lordadamson) [inline selecttool.cpp]: Remove debugging statements before you push please

**Comment by lordadamson** [comment]
Hey man you're doing a really good job over here. I just got a few notes here and their on the code. It should not take a lot of time for you to handle them.

Good Luck and Good job :D

**Comment by 0xbahaa** [comment]
I still can't figure out a way to remove the items from the "savableItems" :(

**Comment by lordadamson** [comment]
Tayeb I've done some stuff on your behalf this time so that you wouldn't be bored to death :D 

Taking a quick look at [my commit](https://github.com/lordadamson/Journal/commit/19cb4b37d484b792e7023ce88d5fda0d69b19f87) you find that I:

I let the paper delete the item by calling `removeItem()` instead of `delete`. because the QGraphicsScene (Paper's superclass) owns that object and should handle the deletion. because it probably does other things besides calling `delete` (like removing it from the list of items or something. so I created a function in Paper that handles deleting items, and in it we did three things:
* We need to notify the `paperModel` so that it would delete it from disk.
* `removeItem(item)` let QGraphicsScene do its thing.
* `savableItems.remove(item);`
* and never forget to set a pointer to null after you delete it. look at [selecttool.cpp:103](https://github.com/lordadamson/Journal/commit/19cb4b37d484b792e7023ce88d5fda0d69b19f87#diff-1cbaf0dcd86664f0938df089ffcd3f6fR103)

**Comment by lordadamson** [comment]
So Good Job man :D Looking forward to your next pull request!

**Pull request merged**